prowler/lib/outputs/jira/jira.py

```diff
@@ -120,14 +126,25 @@ def __init__(
         redirect_uri: str = None,
         client_id: str = None,
         client_secret: str = None,
+        client_mail: str = None,
+        api_token: str = None,
+        site_name: str = None,
     ):
         self._redirect_uri = redirect_uri
         self._client_id = client_id
         self._client_secret = client_secret
+        self._client_mail = client_mail
+        self._api_token = api_token
+        self._site_name = site_name
         self._scopes = ["read:jira-user", "read:jira-work", "write:jira-work"]
-        auth_url = self.auth_code_url()
-        authorization_code = self.input_authorization_code(auth_url)
-        self.get_auth(authorization_code)
+        # If the client mail, api token and site name are present, use basic auth
+        if client_mail and api_token and site_name:
```

Comment:
I think here we should do more validation about the arguments passed, to force each path if the required attributes are set.

Reply:
Done